src/data-pipeline/etl-common/src/main/java/software/aws/solution/clickstream/common/sensors/SensorsEventParser.java

```diff
@@ -0,0 +1,347 @@
+            platform = sensorsEvent.getProperties().getOs().contains("Web") ? "Web" : "Mobile";
+        }
+        clickstreamEvent.setPlatform(platform);
+
+        clickstreamEvent.setProjectId(extraParams.getProjectId());
+
+        clickstreamEvent.setPageViewPageReferrer(deCodeUri(sensorsEvent.getProperties().getReferrer()));
+        clickstreamEvent.setPageViewPageTitle(deCodeUri(sensorsEvent.getProperties().getReferrer_title()));
+        clickstreamEvent.setPageViewPageUrl(deCodeUri(sensorsEvent.getProperties().getUrl()));
+
+        clickstreamEvent.setUserEngagementTimeMsec(sensorsEvent.getProperties().getEvent_duration());
+        clickstreamEvent.setUserId(sensorsEvent.getAnonymous_id());
+        clickstreamEvent.setUserPseudoId(sensorsEvent.getDistinct_id());
+        // clickstreamEvent.setSessionId(sensorsEvent.getGaSessionId());
+        clickstreamEvent.setSessionStartTimeMsec(sensorsEvent.getTime());
+
+        // clickstreamEvent.setSessionNumber(sensorsEvent.getGaSessionNumber());
+        clickstreamEvent.setIp(sensorsEvent.getProperties().getIp());
+        clickstreamEvent.setUa(sensorsEvent.getProperties().getUser_agent());
+
+        // customParameters
+        Map<String, ClickstreamEventPropValue> customParameters = getEventCustomParameters(sensorsEvent, clickstreamEvent);
+        clickstreamEvent.setCustomParameters(customParameters);
+
+        // set traffic source
+        setTrafficSourceBySourceParser(clickstreamEvent);
+
+        Map<String, String> processInfo = new HashMap<>();
+        processInfo.put("rid", extraParams.getRid());
+        processInfo.put("ingest_time", Instant.ofEpochMilli(extraParams.getIngestTimestamp()).toString());
+        processInfo.put(INPUT_FILE_NAME, extraParams.getInputFileName());
+        clickstreamEvent.setProcessInfo(processInfo);
+
+        return clickstreamEvent;
+    }
+
+    private void setDeviceInfo(final SensorsEvent sensorsEvent, final ClickstreamEvent clickstreamEvent) {
+        if (sensorsEvent.getProperties() != null) {
+            clickstreamEvent.setDeviceMobileBrandName(sensorsEvent.getProperties().getProduct_name());
+            clickstreamEvent.setDeviceMobileModelName(sensorsEvent.getProperties().getProduct_classify());
+
+            clickstreamEvent.setDeviceOperatingSystem(sensorsEvent.getProperties().getOs());
+            clickstreamEvent.setDeviceOperatingSystemVersion(sensorsEvent.getProperties().getOs_version());
+        }
+
+        clickstreamEvent.setDeviceSystemLanguage(sensorsEvent.getProperties().getBrowser_language());
+
+        if (sensorsEvent.getProperties() != null && sensorsEvent.getProperties().getUser_agent() != null) {
+            Map<String, Object> deviceUaMap = new HashMap<>();
+            deviceUaMap.put(UA_STRING, sensorsEvent.getProperties().getUser_agent());
+            clickstreamEvent.setDeviceUa(deviceUaMap);
+            Matcher matcher = USER_AGENT_PATTERN.matcher(sensorsEvent.getProperties().getUser_agent());
```

Comment:
do not need to parse UA here, just set `ua` field. there is a enrich plugin to parse `ua` and set them.